joomla/joomla-cms
# [4.0] Allow extensions to add custom data to other extensions in the API

Pull Request for Issue https://github.com/joomla/joomla-cms/issues/26925 .

### Summary of Changes
Adds plugin events to allow extensions to add custom properties to the API Response. Right now this deliberately doesn't allow modification of the existing properties but that will presumably come in the future as the data stabilizes. 

Also has some small cleanups to the Joomla Serializer as I went through this.

### Testing Instructions
[plg_system_demo.zip](https://github.com/joomla/joomla-cms/files/4342894/plg_system_demo.zip)

Install this extension (enabling it etc) and check that the retrieving an article through the API includes extra properties. There should be one relation and one custom property added to the response.

EDIT: I've also tried to doc blocks in the plugin to explain the options and what you can do

### Documentation Changes Required
Needs to be documented how to add these properties (using the plugin as the example)

//cc @nikosdion
